packages/flutter/lib/src/material/text_field.dart

```diff
@@ -1101,6 +1101,7 @@ class _TextFieldState extends State<TextField> with RestorationMixin implements
     _effectiveFocusNode.removeListener(_handleFocusChanged);
     _focusNode?.dispose();
     _controller?.dispose();
+    _controller = null;
```

Comment:
Nulling out in dispose, as a fix for leak, smells.
Here is why: https://github.com/dart-lang/leak_tracker/blob/main/doc/TROUBLESHOOT.md#1-static-or-global-object-causes-notgced-leaks

May be some field on retaining path should be converted to WeakReference?

See other fix for notGCed, that converts a field, not owned by parent, to WeakReference: https://github.com/flutter/flutter/pull/130842